### PR TITLE
Migrate batch routes to use less RouteParams and refactor ConnectionNode

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #33834 

I actually had to revert the initial change of changing the batch changes `<Switch>` to a RR6 `<Routes/>` because the components get mounted inside a RR5 router (user and org routes) and thus it doesn't work. We'll have to migrate these first. 

Some discsusions on the `<ConnectionList />` here: https://sourcegraph.slack.com/archives/C04KDBTLG5P/p1675172882897379

## Test plan

- Manual testing
- CI is ✅

## App preview:

- [Web](https://sg-web-ps-migrate-batch-changes-switch.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.13 kb) | -0.02% (-2.69 kb) | -0.02% (-2.56 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/migrate-batch-changes-switch/compare-report.html;tab=live_object) for a full comparison between the commits 25d8038 and 92daecb or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx [hunk 1]
```diff
@@ -99,58 +100,57 @@ export const NamespaceBatchChangesArea = withAuthenticatedUser<
...
-                        headingElement="h2"
-                        {...props}
-                        {...outerProps}
-                        namespaceID={namespaceID}
-                    />
-                )}
-                exact={true}
-            />
+            <Route path={`${match.url}/apply/:specID`}>
+                <ExtractParams
```
Comment (eseliger) [inline client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx:104]: should we use a type param here? before this used to be typed

**Review by eseliger — APPROVED** [review]

**Comment by philipp-spiess** [comment]
I have no idea why E2E tests are failing now, I can't reproduce this on my local dev instance 🙃

**Comment by philipp-spiess** [comment]
> Maybe networkidle is never happening because of the polling but I don't think I changed anything related to that

Nope that behaves exactly similar to what's deployed on S2 hmm

**Review by valerybugakov — APPROVED** [review]
Let's go!

**Pull request merged**